pkg/clientaccess/clientaccess.go

```diff
@@ -236,14 +236,9 @@ func GetCACerts(u url.URL) ([]byte, error) {
 	u.Path = "/cacerts"
 	url := u.String()
 
-	_, err := get(url, http.DefaultClient, "", "")
-	if err == nil {
-		return nil, nil
-	}
-
 	cacerts, err := get(url, insecureClient, "", "")
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get CA certs at %s", url)
+	if err != nil || cacerts == nil {
```

Comment:
Is it posible for `cacerts` to be nil in the case `err` isn't nil?

Reply:
It looks like we're using `ioutil.ReadAll`, so if `err != nil` the worst we could get is a zero-length byte slice, not nil.

Reply:
possible from 
https://github.com/rancher/k3s/blob/9b1f5bb3d18c48ce328531fc29ddadcd1f4ee935/pkg/clientaccess/clientaccess.go#L264
or 
https://github.com/rancher/k3s/blob/9b1f5bb3d18c48ce328531fc29ddadcd1f4ee935/pkg/clientaccess/clientaccess.go#L255

Reply:
Error would't be nil so the additonal check wouldn't happen.

Reply:
you're right. i remember seeing this case, will double check and remove it if not necessary

Reply:
Right, Brian is asking why you need to check both when it does not appear that it's possible to have a  non-nil err but nil cacerts. All the lines you linked return a non-nil err so the `cacerts == nil` check is unnecessary.

Reply:
after a design discussion with @dweomer and @briandorsey we decided these changes are not needed, and we need to better log these errors 
output: 
```
INFO[0001] Cluster-Http-Server2020/09/14 19:59:13 http: TLS handshake error from 127.0.0.1:58266: remote error: tls: bad certificate 
INFO[0001] Cluster-Http-Server2020/09/14 19:59:13 http: TLS handshake error from 127.0.0.1:58274: remote error: tls: bad certificate 
```